### PR TITLE
Issue 32 initial commit (#1321)

### PR DESCRIPTION
**Pull request by thefazzer**
**Issue 32 initial commit (#1321)**

- Call FullValidationSucceeded on some header.
- Make sure header.ValidationState == FV.

### CODE REVIEW

**Comment by Aprogiena** [comment]
Looks good to me

File: src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs [hunk 1]
```diff
@@ -1284,5 +1284,38 @@ public void ChainWithMaxReorgPlusExtraHeadersIsCalled_AnotherChainIsPresented_Co
...
+                chainedHeaderTree.ConnectNewHeaders(1, listOfChainHeaders);
+            chainTip = connectNewHeadersResult.Consumed;
+
+            // Select an arbitrary header h19 on the extended chain.
+            ChainedHeader newHeader = chainTip.GetAncestor(19);
+
+            // Verify its initial BlockValidationState.
+            Assert.Equal(ValidationState.HeaderValidated, newHeader.BlockValidationState);
+
+            // Call FullValidationSucceeded and verify BlockValidationState has changed.
```
Comment (majikandy) [inline src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs]: This comment is a repeat of the code below it, please remove

Comment (thefazzer) [reply src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs]: Done

**Comment by thefazzer** [comment]
@noescape00 could you take a look & merge if OK pls

**Comment by Aprogiena** [comment]
Now there is no code again

**Comment by noescape00** [comment]
Conflicts

**Comment by thefazzer** [comment]
Fixed conflicts.

**Pull request merged**